[Eliot Lee Leong Wern] iP

eliotlee68 opened the pull request:
# AaronBot
> NUS Teaching Excellence Honour Roll (2010)
>NUS Annual Teaching Excellence Award (2001/2, 2002/3, 2007/8, 2008/9)
>SoC Teaching Excellence Honour Roll (2009)
>SoC Teaching Excellence Award (1999-2009)
>Outstanding Mentor Award, Science Mentorship Programmes (2008)
>Students' Choice Award (2004/5, 2005/6)
>[source](https://www.comp.nus.edu.sg/cs/people/tantc/)
### **NOTE: This is NOT the real prof Aaron**
That's right student, It's me, Prof Aaron (condensed into the form of a Java chatbot) 😀 
What am I doing as a chatbot you ask? Well I'm here to help you keep track of your tasks!
Features include:
- Adding *3* types of tasks (TODO, DEADLINE, and EVENT)
- Marking/unmarking tasks as done
- Deleting tasks
- Saving/loading task lists from local storage
- Searching for tasks by keyword

All you need to do is
1. Download it
2. Locate `main/java/aaron/main/AaronBot.java`
3. Run it
4. That's it student !

Features:
[X] Managing tasks
[ ] Reminders (Coming soon!)
[ ] JSON App to launch (Coming soon!)

alternatively, you can run it on java: 
``` ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

leeliot68 committed:
level 0

leeliot68 committed:
Implemented level 1.

leeliot68 committed:
Implemented task list feature`

leeliot68 committed:
Implemented level 3.

leeliot68 committed:
Implement level 3

AaronBot.java: Updated initial instructions, changed chat function to
use switch-case instead of if-else, changed addtolist to account for
different types of tasks
Task class: Override equals() method to check for duplicates
Added Event, Deadline and Todo classes that extend from task class.

leeliot68 committed:
Added A-TextUITesting

leeliot68 committed:
Implemented level 5 error checking and handling.

leeliot68 committed:
Implemented level 6, delete tasks functionality.

leeliot68 committed:
Added javadocs as well as changed behaviours of certain functions to
accurately reflect desired behaviour

leeliot68 committed:
Implemented level 7, write to file and load from file funcitonality.

eliotlee68 committed:
Merge pull request #1 from eliotlee68/branch-Level-7

Implemented level 7, write to file and load from file functionality.

jannaleong commented on src/main/java/AaronBot.java:
```diff
@@ -0,0 +1,275 @@
+     * @return boolean value representing whether the chat should still be continued
+     * @throws NonsenseCommandException if command given by user is not recognized by Aaronbot
+     */
+    private static boolean chat(String userInput) throws NonsenseCommandException{
```
consider using a plural noun instead of tokenizedUserInput as your string array contains multiple strings.

jannaleong commented on src/main/java/AaronBot.java:
```diff
@@ -0,0 +1,275 @@
+        default:
+            throw new NonsenseCommandException(userInput);
+        }
+    }
```
consider removing this comment as it does not improve the functionality of your code.

jannaleong commented on src/main/java/TaskListFile.java:
```diff
@@ -0,0 +1,90 @@
+        fileScanner.close();
+        return loadTaskList;
+    }
+
```
Consider removing this comment below.

jannaleong reviewed (commented):
Overall, very minimal errors in naming your methods and variables. Good job!! Just some small errors here and there.

jannaleong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,63 @@
+/**
```
Good job!! all naming conventions are being followed in this file.

jannaleong reviewed (commented):
Wanted to affirm you that the other files are following the naming conventions well.

leeliot68 committed:
Implemeneted level 8: Deadline and Event only accept specific dateTime
formats.

eliotlee68 committed:
Merge pull request #2 from eliotlee68/branch-Level-8

Implemeneted level 8: Deadline and Event only accept specific dateTime

leeliot68 committed:
Implemented A-MoreOOP, abstracted functionality into various classes and
runs from the main AaronBot.java file

eliotlee68 committed:
Merge pull request #3 from eliotlee68/branch-A-MoreOOP

Implemented A-MoreOOP, abstracted functionality into various classes and

leeliot68 committed:
Sorted files into packages, no functionality changed.

eliotlee68 committed:
Merge pull request #4 from eliotlee68/Branch-A-Packages

Sorted files into packages, no functionality changed.

leeliot68 committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support' into Branch-A-Gradle

eliotlee68 committed:
Merge pull request #5 from eliotlee68/Branch-A-Gradle

Branch a gradle

leeliot68 committed:
Added JUnit testing files

eliotlee68 committed:
Merge pull request #6 from eliotlee68/Branch-A-JUnit

Added JUnit testing files

rizkidelta commented on src/main/java/Aaron/Main/AaronBot.java:
```diff
@@ -0,0 +1,50 @@
+package Aaron.Main;
```
Names representing packages should be in all lower case.

rizkidelta commented on src/main/java/Aaron/UI/UI.java:
```diff
@@ -0,0 +1,67 @@
+import Aaron.Exception.AaronBotException;
+import Aaron.Task.TaskList;
+public class UI {
+    private static final Scanner scanner = new Scanner(System.in);
```
Nit: Constant names must be all uppercase

rizkidelta commented on src/main/java/Aaron/UI/UI.java:
```diff
@@ -0,0 +1,67 @@
+        }
+    }
+
+    public void showParseError(String userString) {
+        System.out.println("Sorry student, we had an error parsing message: \n"
+                + userString);
+    }
+
+    public void markMessage(int taskIndex, TaskList taskList) {
+        System.out.println("Successfully marked task: \n"
+                + taskList.printTask(taskIndex));
+    }
+
+    public void unmarkMessage(int taskIndex, TaskList taskList) {
+        System.out.println("Successfully unmarked task: \n"
+                + taskList.printTask(taskIndex));
+    }
+
+    public void taskAddedMessage(TaskList taskList) {
+        System.out.println("Successfully added task: \n"
+                + taskList.printTask(taskList.getTasklistSize()));
+    }
+
+    public void errorMessage(AaronBotException e) {
+        System.out.println("Oh no, there seems to have been an error");
+        System.out.println(e);
+    }
```
I like that you followed the indentation for wrapped lines (8spaces)

rizkidelta commented on src/main/java/Aaron/Task/Task.java:
```diff
@@ -0,0 +1,62 @@
+    private boolean isDone;
+    private String taskString;
+
+    public Task(String taskString) throws TaskNoNameException {
+        this.isDone = false;
+        if (taskString.trim().isEmpty()) {
+            throw new TaskNoNameException("STUDENT!! You need a task description :)");
+        } else {
+            this.taskString = taskString;
+        }
+    }
+
+    public Task(String taskString, boolean isDone) {
+        this.isDone = isDone;
+        this.taskString = taskString;
+    }
+
+    public void markDone() throws DoubleMarkException{
+        if (isDone) {
+            throw new DoubleMarkException("Task is already marked \n " + this.toString());
+        } else { 
+            isDone = true;
+        }
+    }
+
+    public boolean unmarkDone() throws DoubleMarkException{
+        if (!isDone) {
+            throw new DoubleMarkException("Task is already unmarked \n " + this.toString());
+        } else {
+            isDone = false;
+            return true;
+        }
+    }
+
```
I like that you have custom exceptions which are named appropriately following the convention of ending with 'Exception'

rizkidelta commented on src/main/java/Aaron/Storage/Storage.java:
```diff
@@ -0,0 +1,88 @@
+ * @throws IOException
+ */
+public class Storage {
+    public static void writeToFile(String filePath, TaskList taskListWrite) throws IOException{
+        Path path = Paths.get(filePath);
+        Files.createDirectories(path.getParent());
+        File prevTaskList = new File(filePath);
+        prevTaskList.createNewFile();
+        if (prevTaskList.exists()) {
+            prevTaskList.delete();
+        }
+        try {
+            FileWriter writer = new FileWriter(filePath, true);
+            for (int i = 1; i < taskListWrite.getTasklistSize() + 1; i++) {
+                String taskDesc = taskListWrite.printTask(i).toString();
+                writer.write(taskDesc + System.lineSeparator());
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new IOException("Error loading file");
+        }
+    }
+
```
In the writeToFile method, the use of FileWriter in append mode (new FileWriter(filePath, true)) is highlighted as redundant due to the explicit file deletion logic. Append mode inherently handles the creation of a new file if it doesn't exist, and appends data to the end of an existing file. Therefore, the manual deletion of the file prior to writing is unnecessary and could be removed to streamline the process.

rizkidelta reviewed (commented):
Overall, the code is very well-written. I particularly appreciate the use of object-oriented programming principles, evident in the separation of classes to function independently. Aside from the common oversight of not using lowercase for package names and a few minor logical issues, I would say it's an excellent effort.

leeliot68 committed:
Added javadocs for methods and class headers

leeliot68 committed:
Added JavaDocs for class and methods

leeliot68 committed:
Edited file to follow coding conventions

leeliot68 committed:
Implemented level 9: Search function

leeliot68 committed:
Merge branch 'Branch-A-CodingStandard'

leeliot68 committed:
Merge branch 'Branch-Level-9'

leeliot68 committed:
Cleaned up errors from merging branches

leeliot68 committed:
Added resources folder

leeliot68 committed:
Added GUI Support with JavaFX

leeliot68 committed:
Merge branch 'Branch-level-10'

leeliot68 committed:
Added missing javadocs for methods

leeliot68 committed:
Bug fixes with storage and fxml changes

leeliot68 committed:
Added assertions for certain methods

leeliot68 committed:
Update code to adhere to coding standards

Command methods have a method returnIsBye which is not needed anymore,
similarly the BYE command is not needed anymore as MainWindow.java
handles termination condition.

Delete aforementioned methods and class.

Adding javadocs for all methods and classes

TaskList.java: refactoring addtolist method to improve readability

Storage.java: refactoring loadfromfile method to improve readability

eliotlee68 committed:
Merge pull request #8 from eliotlee68/Branch-A-Assertions

Added assertions for certain methods

leeliot68 committed:
Merge branch 'Branch-A-CodingQuality'

leeliot68 committed:
Implemented BCD-Extension B-Snooze

Add Snooze.java file to implement logic for snoozing tasks where run
postpones a task by 3 days

TaskList.java: Add logic to postpone a specific task

Task.java: Add abstract postpone method

Event.java: postpone method postpones start and end date by n days

Deadline.java: postpone method postpones deadline by n days

Task.java: throw aaronbotexception as nothing to postpone

CommandTypes.java: Add Snooze function to enum

eliotlee68 committed:
Merge pull request #10 from eliotlee68/Branch-BCD-Extension

Implemented BCD-Extension B-Snooze

leeliot68 committed:
Modified UI interactions to include personality

leeliot68 committed:
Added user guide and image

leeliot68 committed:
Edited Readme user guide
